tests/recipes/cad1/task1/test_enhance.py

```diff
@@ -0,0 +1,253 @@
+        ]
+    )
+    ref = np.array([4.0, 5.0, 6.0])
+    expected_result = [
+        np.array([[4.0, 5.0, 6.0], [4.0, 5.0, 6.0], [4.0, 5.0, 6.0]]),
+        np.array([[5.0, 5.0, 5.0], [5.0, 5.0, 5.0], [5.0, 5.0, 5.0]]),
+    ]
+    result = denormalize_signals(sources, ref)
+    assert len(result) == len(expected_result)
+    assert np.allclose(result[0], expected_result[0])
+    assert np.allclose(result[1], expected_result[1])
+
+    # Test case 2: Test the function with an input that has a non-zero mean
+    sources = np.array(
+        [
+            np.array(
+                [
+                    [-1.22474487, 0.0, 1.22474487],
+                    [-1.22474487, 0.0, 1.22474487],
+                    [-1.22474487, 0.0, 1.22474487],
+                ]
+            ),
+            np.array([[0.0, 0.0, 0.0], [0.0, 0.0, 0.0], [0.0, 0.0, 0.0]]),
+        ]
+    )
+    ref = np.array([1.0, 2.0, 3.0])
+    expected_result = [
+        np.array([[1.0, 2.0, 3.0], [1.0, 2.0, 3.0], [1.0, 2.0, 3.0]]),
+        np.array([[2.0, 2.0, 2.0], [2.0, 2.0, 2.0], [2.0, 2.0, 2.0]]),
+    ]
+    result = denormalize_signals(sources, ref)
+    assert len(result) == len(expected_result)
+    assert np.allclose(result[0], expected_result[0])
+    assert np.allclose(result[1], expected_result[1])
+
+    # Test case 3: Test the function with an input that has a non-unit variance
+    sources = np.array(
+        [
+            np.array(
+                [
+                    [-2.44948974, 0.0, 2.44948974],
+                    [-2.44948974, 0.0, 2.44948974],
+                    [-2.44948974, 0.0, 2.44948974],
+                ]
+            ),
+            np.array([[0.0, 0.0, 0.0], [0.0, 0.0, 0.0], [0.0, 0.0, 0.0]]),
+        ]
+    )
+    ref = np.array([4.0, 5.0, 6.0])
+    expected_result = [
+        np.array([[3.0, 5.0, 7.0], [3.0, 5.0, 7.0], [3.0, 5.0, 7.0]]),
+        np.array([[5.0, 5.0, 5.0], [5.0, 5.0, 5.0], [5.0, 5.0, 5.0]]),
+    ]
+    result = denormalize_signals(sources, ref)
+    assert len(result) == len(expected_result)
+    assert np.allclose(result[0], expected_result[0])
+    assert np.allclose(result[1], expected_result[1])
+
+
+def test_map_to_dict(regtest):
+    sources = np.array([[1, 2], [3, 4], [5, 6]])
+    sources_list = ["a", "b", "c"]
+    output = map_to_dict(sources, sources_list)
+    regtest.write(f"output: \n{output}\n")
+
+
+def test_decompose_signal(regtest):
+    np.random.seed(123456789)
+    # Load Separation Model
+    model = HDEMUCS_HIGH_MUSDB.get_model().double()
+    device = torch.device("cpu")
+    model.to(device)
+
+    # Create a mock signal to decompose
+    fs = 44100
+    duration = 5
+    signal = np.random.uniform(size=(1, 2, fs * duration))
+    # Call the decompose_signal function and check that the output has the expected keys
+    output = decompose_signal(model, signal, fs, device)
+    regtest.write(f"output: \n{output}\n")
+
+
+def test_apply_baseline_ha(regtest):
```

Comment:
I don't think this needs to be a `regtest` either. The difference between a `unittest` and a `regtest` is that the former is testing a single function, possibly with variable input parameters (as you have parametrized the above tests). A regression tests tests multiple steps in a processing chain that are composed of many separate classes and methods/functions all depending on each other.